### ISSUE TITLE
 'str' object has no attribute 'type'

### ISSUE THREAD

**Issue by Sloppyedwards** [wordfence/wordfence-cli#119]
Installed wordfence using "pip install wordfence" on a NearlyFreeSpeech shared hosting site. Wordfence installed to /home/private/.local/bin/wordfence.

Command: [username /home/public]$ wordfence vuln-scan ./ -d

Output:
Scanning site at ./...
Located WordPress files at ./
WordPress Core Version: 6.3.2
Traceback (most recent call last):
  File "/home/private/.local/bin/wordfence", line 8, in <module>
    sys.exit(main())
  File "/home/private/.local/lib/python3.10/site-packages/wordfence/cli/cli.py", line 181, in main
    exit_code = cli.invoke()
  File "/home/private/.local/lib/python3.10/site-packages/wordfence/cli/cli.py", line 175, in invoke
    return self.process_exception(exception)
  File "/home/private/.local/lib/python3.10/site-packages/wordfence/cli/cli.py", line 102, in process_exception
    raise exception
  File "/home/private/.local/lib/python3.10/site-packages/wordfence/cli/cli.py", line 171, in invoke
    return self.subcommand.invoke()
  File "/home/private/.local/lib/python3.10/site-packages/wordfence/cli/vulnscan/vulnscan.py", line 216, in invoke
    self._scan_site(
  File "/home/private/.local/lib/python3.10/site-packages/wordfence/cli/vulnscan/vulnscan.py", line 153, in _scan_site
    self._scan(
  File "/home/private/.local/lib/python3.10/site-packages/wordfence/cli/vulnscan/vulnscan.py", line 86, in _scan
    self._scan_plugins(site.get_all_plugins(), scanner)
  File "/home/private/.local/lib/python3.10/site-packages/wordfence/wordpress/site.py", line 275, in get_all_plugins
    plugins = self.get_plugins(mu=True)
  File "/home/private/.local/lib/python3.10/site-packages/wordfence/wordpress/site.py", line 254, in get_plugins
    for path in self._generate_possible_plugins_paths(mu):
  File "/home/private/.local/lib/python3.10/site-packages/wordfence/wordpress/site.py", line 241, in _generate_possible_plugins_paths
    configured = self.get_configured_plugins_directory(mu)
  File "/home/private/.local/lib/python3.10/site-packages/wordfence/wordpress/site.py", line 233, in get_configured_plugins_directory
    return self._extract_string_from_config(
  File "/home/private/.local/lib/python3.10/site-packages/wordfence/wordpress/site.py", line 188, in _extract_string_from_config
    state = self._get_parsed_config_state()
  File "/home/private/.local/lib/python3.10/site-packages/wordfence/wordpress/site.py", line 179, in _get_parsed_config_state
    self.config_state = self._parse_config_file()
  File "/home/private/.local/lib/python3.10/site-packages/wordfence/wordpress/site.py", line 166, in _parse_config_file
    return context.evaluate(
  File "/home/private/.local/lib/python3.10/site-packages/wordfence/php/parsing.py", line 875, in evaluate
    instruction.evaluate(state)
  File "/home/private/.local/lib/python3.10/site-packages/wordfence/php/parsing.py", line 593, in evaluate
    new_value = component.evaluate(state)
  File "/home/private/.local/lib/python3.10/site-packages/wordfence/php/parsing.py", line 751, in evaluate
    arguments = [argument.evaluate(state) for argument in self.arguments]
  File "/home/private/.local/lib/python3.10/site-packages/wordfence/php/parsing.py", line 751, in <listcomp>
    arguments = [argument.evaluate(state) for argument in self.arguments]
  File "/home/private/.local/lib/python3.10/site-packages/wordfence/php/parsing.py", line 603, in evaluate
    value = operator.apply(value, new_value)
  File "/home/private/.local/lib/python3.10/site-packages/wordfence/php/parsing.py", line 518, in apply
    return self.callable(left, right)
  File "/home/private/.local/lib/python3.10/site-packages/wordfence/php/parsing.py", line 531, in <lambda>
    lambda left, right: PhpValue(left.type, left.value + right.value)
AttributeError: 'str' object has no attribute 'type'

Same output if I use "wordfence vuln-scan . -d" or "wordfence vuln-scan /home/public/ -d"

Am I doing something wrong?  What other info do you need to troubleshoot this error?

**Comment by akenion** [comment]
This is an issue with how CLI parses WordPress' `wp-config.php` file to identify any non-standard path configurations. I was able to trigger the same trace using a construct like this in the wp-config file:
```php
define('SOME_CONSTANT', ANOTHER_CONSTANT . '/some-string');
```

I assume you have something similar (a constant concatenated with a string) in your wp-config file. If that is the case, I have a fix ready (#120) and will include it in the next CLI release.

**Comment by Sloppyedwards** [comment]
Yes, looks to be the same issue.  Towards the end of my wp-config.php file are the following lines:

```
define('WP_TEMP_DIR',ABSPATH.'tmp');

/** Absolute path to the WordPress directory. */
if ( !defined('ABSPATH') )
        define('ABSPATH', dirname(__FILE__) . '/');

/** Sets up WordPress vars and included files. */
require_once(ABSPATH . 'wp-settings.php');

```

I didn't put them there, so I'm not sure if they are standard or _were_ standard under some past version.  Thanks for the update!  I'll watch for the next release and try again.
